src/crypto/secretstream/xchacha20poly1305.rs

```diff
@@ -45,7 +45,7 @@ mod test {
 
     #[test]
     fn decrypt_too_short_ciphertext() {
-        let ciphertext: [u8; (ABYTES - 1)] = unsafe { mem::uninitialized() };
+        let ciphertext = [0; (ABYTES - 1)];
```

Comment:
This is less unsafe but you just remove unsafe and make the code more slow. It's a choice but I think you mix the clear goal to replace `uninitialized` by `MaybeUninit` and the goal to remove unsafe at all cost. This should be in a separate PR I think.

Reply:
It's in test code, so performance is not an issue.